amplify/src/main/java/com/github/stkent/amplify/tracking/Amplify.java

```diff
@@ -223,70 +241,44 @@ public Amplify setPackageName(@NonNull final String packageName) {
 
     @Override
     public void notifyEventTriggered(@NonNull final IEvent event) {
-        logger.d(event + " event triggered");
+        logger.d(event.getTrackingKey() + " event triggered");
         totalEventCountRulesManager.notifyEventTriggered(event);
         firstEventTimeRulesManager.notifyEventTriggered(event);
         lastEventTimeRulesManager.notifyEventTriggered(event);
         lastEventVersionCodeRulesManager.notifyEventTriggered(event);
         lastEventVersionNameRulesManager.notifyEventTriggered(event);
-    }
 
-    // Query methods
+        if (event == PromptInteractionEvent.USER_GAVE_POSITIVE_FEEDBACK) {
+            final Activity activity = activityReferenceManager.getValidatedActivity();
 
-    // todo: track activity using application hooks so we don't need to receive it here
-    public void promptIfReady(
-            @NonNull final Activity activity,
-            @NonNull final IPromptView promptView) {
+            if (activity != null) {
```

Comment:
`getValidatedActivity` returns null if either of those conditions is true, so this check is equivalent!